internal/pkg/agent/cmd/enroll_cmd_test.go

```diff
@@ -424,6 +421,79 @@ func TestEnroll(t *testing.T) {
 			assert.Equal(t, host, config.Client.Host)
 		},
 	))
+
+	t.Run("headers are sent to server", withServer(
+		func(t *testing.T) *http.ServeMux {
+			mux := http.NewServeMux()
+			mux.HandleFunc("/api/fleet/agents/enroll", func(w http.ResponseWriter, r *http.Request) {
+				if r.Header.Get("X-Test-Header") != "Test-Value" {
```

Comment:
This proves we use a header for `/enroll`, is this enough to also prove the same header would be included in a checkin?

I see in enroll.go we explicitly add the header to `enrollCmdOption` so it isn't obvious just by reading it. Regardless, would there be a way to test that?